From fd6a33c2ae063f51fd40da12a61543cefa77aa41 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 25 Oct 2023 17:08:31 +0200
Subject: [PATCH 0295/1695] drivers: stm32_serc: add
 CFG_STM32_PANIC_ON_SERC_EVENT switch

Do panic() when receiving a SERC interrupt only if
CFG_STM32_PANIC_ON_SERC_EVENT is set. It is default disable to allow
the data abort generated along with the SERC event to be caught.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6b3539e2042613ffa03f050d3832007d7ebc73b6 drivers: stm32_serc: add CFG_STM32_PANIC_ON_SERC_EVENT switch]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: Ie7a619cc38346d69dc7687e6dc3e86d96ef10ced
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/336213
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 5 +++++
 core/drivers/firewall/stm32_serc.c  | 3 ++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 2a1ec3983..842815927 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -94,3 +94,8 @@ endif
 
 # Optional behavior upon receiving illegal access events
 CFG_STM32_PANIC_ON_IAC_EVENT ?= y
+ifeq ($(CFG_TEE_CORE_DEBUG),y)
+CFG_STM32_PANIC_ON_SERC_EVENT ?= n
+else
+CFG_STM32_PANIC_ON_SERC_EVENT ?= y
+endif
diff --git a/core/drivers/firewall/stm32_serc.c b/core/drivers/firewall/stm32_serc.c
index 01bfc2290..ef902a3e9 100644
--- a/core/drivers/firewall/stm32_serc.c
+++ b/core/drivers/firewall/stm32_serc.c
@@ -145,7 +145,8 @@ static enum itr_return stm32_serc_itr(struct itr_handler *h __unused)
 		}
 	}
 
-	stm32_rif_access_violation_action();
+	if (IS_ENABLED(CFG_STM32_PANIC_ON_SERC_EVENT))
+		stm32_rif_access_violation_action();
 
 	return ITRR_HANDLED;
 }
-- 
2.47.2

